### PR TITLE
Authentification integrated in develop branch

### PR DESCRIPTION
**Pull request by nicolasp9-dev**

### CODE REVIEW

**Comment by Jmion** [comment]
You might want to look at AuthentificationProcess.java:213
the logging tag can be at most 23 characters, was 27.
It appears that this error has been in a few travis details log. This might help you get closer to a working travis build

**Comment by nicolasp9-dev** [comment]
I submitted a new version without TAG warning

File: app/src/main/java/ch/epfl/sweng/favours/AuthentificationProcess.java [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+        }
+
+        if (status == FavoursMain.Status.Login) {
+            mAuth.signInWithEmailAndPassword(email, password)
+                    .addOnCompleteListener(this, new OnCompleteListener<AuthResult>() {
+                        @Override
+                        public void onComplete(@NonNull Task<AuthResult> task) {
+                            if (task.isSuccessful()) {
+                                RuntimeEnvironment.getInstance().isConnected.set(true);
+                                Log.d(TAG, "signInWithEmail:success");
```
Comment (Jmion) [inline app/src/main/java/ch/epfl/sweng/favours/AuthentificationProcess.java]: This is where the tag is to long

**Comment by Jmion** [comment]
Issues with the tests. The error that was causing issues previously is gone

File: .idea/misc.xml [hunk 2]
```diff
@@ -5,27 +5,31 @@
...
           <item index="2" class="java.lang.String" itemvalue="edu.umd.cs.findbugs.annotations.NonNull" />
           <item index="3" class="java.lang.String" itemvalue="android.support.annotation.NonNull" />
+          <item index="4" class="java.lang.String" itemvalue="androidx.annotation.NonNull" />
+          <item index="5" class="java.lang.String" itemvalue="androidx.annotation.RecentlyNonNull" />
         </list>
       </value>
     </option>
   </component>
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" project-jdk-name="1.8" project-jdk-type="JavaSDK">
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_7" project-jdk-name="1.8" project-jdk-type="JavaSDK">
```
Comment (Jmion) [inline .idea/misc.xml]: Why do wee need to have the version number at 1_7?

Comment (nicolasp9-dev) [reply .idea/misc.xml]: I have no idea why it changed it

Comment (nicolasp9-dev) [reply .idea/misc.xml]: Misc files are autogenerated by the project and seems to change depending of libraries linked to it, atm I'm writing Android Studio changed it back to JDK_1_8

**Pull request merged**